easybuild/easyblocks/c/cppcheck.py

```diff
@@ -0,0 +1,141 @@
+        extra.update(extra_vars)
+        return ConfigureMake.extra_options(extra_vars=extra)
+
+    def configure_step(self):
+        """
+        Run qmake on the GUI, if necessary
+        """
+        if self.cfg['build_gui']:
+            if self.cfg['have_rules']:
+                cmd = 'qmake HAVE_RULES=yes'
+            else:
+                cmd = 'qmake'
+            
+            try:
+                gui_dir = os.path.join(self.cfg['start_dir'], 'gui')
+                os.chdir(gui_dir)
+                run_cmd(cmd, log_all=True, simple=True)
+                os.chdir(self.cfg['start_dir'])
+            except OSError as err:
+                raise EasyBuildError("Moving to %s and configure the GUI build failed: %s", gui_dir, err)
+
+        else:
+            self.log.debug("Configuration of the GUI skipped")
+            pass
+
+    def build_step(self, verbose=False):
+        """
+        Compile cppcheck with make and cppcheck-gui with qmake and make 
+        """
+        self.cfg.update('buildopts', 'CFGDIR=%(installdir)s/cfg')
+        
+        if self.cfg['have_rules']:
+            self.cfg.update('buildopts', 'HAVE_RULES=yes')
+
+        super(EB_cppcheck, self).build_step(verbose=verbose)
+        
+        if self.cfg['build_gui']:
+            super(EB_cppcheck, self).build_step(verbose=verbose, path='gui')
+
+    def install_step(self):
+        """
+        Install cppcheck with make install and cppcheck-gui copying the file
+        """
+        self.cfg.update('installopts', 'DESTDIR=%(installdir)s/ PREFIX="" CFGDIR=cfg')
+        
+        super(EB_cppcheck, self).install_step()
+       
+        # The GUI doesn't have make install, so we copy it manually
+        if self.cfg['build_gui']:
+            try:
+                # make sure we're (still) in the start dir
+                os.chdir(self.cfg['start_dir'])
+
+                file_to_copy = 'cppcheck-gui'
+
+                filepath = os.path.join(self.cfg['start_dir'], 'gui/', file_to_copy)
+
+                # Perform the copy
+                target_dest = os.path.join(self.installdir, 'bin')
+                if os.path.isfile(filepath):
+                    self.log.debug("Copying file %s to %s" % (filepath, target_dest))
+                    shutil.copy2(filepath, target_dest)
+                else:
+                    raise EasyBuildError("Can't copy non-existing path %s to %s", filepath, target_dest)
+                
+                # Create cfg symlink. It shouldn't be necessary, but cppcheck-gui complaints otherwise
+                # unless called with --data-dir
+                src = os.path.join(self.installdir, 'cfg')
+                dst = os.path.join(self.installdir, 'bin/cfg')
+                self.log.debug("Creating symlink %s to %s" % (src, dst))
+                os.symlink(src, dst)
+
+            except OSError as err:
+                raise EasyBuildError("Copying %s to installation dir failed: %s", file_to_copy, err)
```

Comment:
drop `file_to_copy`, use `filepath` here instead (and define `filepath` outside of the `try`